codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -905,13 +905,16 @@ private static int getChunkSize(byte[] hex, int start, int length) {
                         // empty case
                         throw new NumberFormatException();
                     }
-                    return result;
+                    break;
                 }
                 // non-hex char fail-fast path
                 throw new NumberFormatException();
```

Comment:
While you are here, consider adding exception messages for other `NumberFormatException` cases